### PR TITLE
[Doc] Add issue template

### PR DESCRIPTION
**Pull request by AmirAbrams**
#### What is the purpose of this pull request (PR)?

Adds an issue template.
#### Any background context to help the reviewer?

Issue templates are new to Github and not yet implemented in this repo: [Github issue-template](https://github.com/blog/2111-issue-and-pull-request-templates)
### Describe the issue

This is an example issue.
#### List steps to reproduce below:
1.  Open Qt wallet
2.  Open Menu...
3.  Press test button
### Expected behavior

This is just an example.
### Actual behavior

This is just an example.
### Any information in the debug.log file related to this issue?

Example data:
`2016-08-10 05:16:25 keypool added key 135, size=135`
### Screenshots (if available)

N/A
### What binary version was used (official or self compiled)

official binary version 0.12.
### Machine specs: CPU, RAM, Disk space & OS (Windows, OS X, Linux)

Linux (Ubuntu) 16.04 64 bit, 2x vCPU, 2 TB disk and 6 GB RAM.

### CODE REVIEW

**Comment by btcdrak** [comment]
We definitely need something like this for issues so people can more accurately report bugs.

**Comment by AmirAbrams** [comment]
Added an issue template as well.

**Comment by jonasschnelli** [comment]
I think we should add this to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md (this page is referred to while creating an issue or PR).

**Comment by MarcoFalke** [comment]
@jonasschnelli I am assuming GitHub will auto-fill the body of the pull or issue with this template, so you don't need to refer to it.

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#### What is the purpose of this pull request (PR)?
+
+#### Any background context to help the reviewer?
+
+#### Was this PR tested and how?
+
+#### Does this PR resolve an open issue?
```
Comment (MarcoFalke) [inline PULL_REQUEST_TEMPLATE.md]: We usually put `Fixes #----` in the body of the pull to indicate that an open issue is resolved. Not sure if having a section for this in every pull makes sense.

**Comment by AmirAbrams** [comment]
I will add the Fixes # to the PR template and squash my commits into one.

Is this something you want to implement (ACK or NACK)?

**Comment by fanquake** [comment]
@AmirAbrams It is something worth adding, but make the fixes you've mentioned before asking for ACK/NACKS.

**Comment by AmirAbrams** [comment]
Okay, I made the requested change and squashed my commits into one.

**Comment by arowser** [comment]
Can one of the admins verify this patch?

**Comment by fanquake** [comment]
A few thoughts.
PR Template:
- mention adding [skip-ci] so that trivial/doc only changes don't get run on Travis.
- You can probably remove the more-questions section
- Add a note about adding `[doc]` or `[RPC]` etc to the title.
- Screenshots can be added, they are necessary for GUI changes.
- Could have a `what machines/os have you tested/compiled this code?`

Issue Template:
- What version of bitcoin-core are you using? Official release or self-compiled?
- Basic specs, such as disk space and available ram.

**Comment by MarcoFalke** [comment]
> mention adding [skip-ci] so that trivial/doc only changes don't get run on Travis.

It now has to be somewhere in the commit body instead of just the GitHub pull. Taking into account that travis is now pretty fast due to efficient caching, it may not be worth to mention it in the template.

In general I am not in favor of adding too much to the pull request template, because stuff that is not relevant has to be deleted every time a pull is created. I may even prefer no pull request template at all.

**Comment by AmirAbrams** [comment]
@fanquake, I will make your requested changes, move the templates to a hidden `.github `directory and squash my commits but don't want to waist time if the Bitcoin team does not want to implement these github templates.

@MarcoFalke, please let me know how you want to proceed with this PR.

Thanks!

**Comment by MarcoFalke** [comment]
I think having an issue template makes sense to preempt https://github.com/bitcoin/bitcoin/issues/8130#issuecomment-223238405 et al.

I am not convinced about the pull template, but I am happy to hear other opinions. (Preferably by people who regularly create pulls)

**Comment by paveljanik** [comment]
I do not think PR template is needed. But for Issue, it is OK!

**Comment by fanquake** [comment]
@AmirAbrams Consensus seems to be to add just an issue template, can you update this pull?

**Comment by AmirAbrams** [comment]
Yes, I will update the commit with your changes.

**Comment by MarcoFalke** [comment]
Nit: Our preference is to use present-tense, imperative-style commit messages. (The default for git)

**Comment by MarcoFalke** [comment]
utACK 3897668

**Comment by laanwj** [comment]
utACK 3897668

**Pull request merged**